dts/bindings/adc/nuvoton,adc-cmp.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    This will perform signal comparision with threshold established.
+
+compatible: "nuvoton,adc-cmp"
+
+include : adc-comparator.yaml
+
+properties:
+    threshold-mv:
+      type: int
+      required: false
+      description: |
+        Value in millivolts present on ADC data as threshold assert.
+
+    comparison:
+      type: string
+      required: false
+      description: |
+        Determines the condition to be met between ADC data and
+        threshold assert value that will trigger comparator event.
+      enum:
+        - ADC_CMP_NPCX_GREATER
+        - ADC_CMP_NPCX_LESS_OR_EQUAL
```

Comment:
I have been thinking about changing this a little bit to remove some ambiguity, current yaml yields to the following dts:
        adc-cmp {
                adc-cmp1 {
                        compatible = "nuvoton,adc-cmp";
                        io-channels = <&adc0 3>;
                        comparison = "ADC_CMP_NPCX_GREATER";
                        threshold-mv = <2000>;
                };
                adc-cmp2 {
                        compatible = "nuvoton,adc-cmp";
                        io-channels = <&adc0 5>;
                        comparison = "ADC_CMP_NPCX_LESS_OR_EQUAL";
                        threshold-mv = <500>;
                };

Having to repeat 'compatible', and &adc0 on each io-channles property.

Perhaps something to look more like this: 
        adc-cmp {
                compatible ="nuvoton,adc-cmp"
                adc = &adc0;
                /* Include any parameter shared across all adc-cmpX entities like 'adc' */
                adc-cmp1 {
                        channel = <3>;
                        comparison = "ADC_CMP_NPCX_GREATER";
                        threshold-mv = <2000>;
                };
                adc-cmp2 {
                        channel = <5>;
                        comparison = "ADC_CMP_NPCX_LESS_OR_EQUAL";
                        threshold-mv = <500>;
                };
        };

Please let me know your thoughts. 

Also, by changing this, maybe dts/bindings/adc/adc-comparator.yaml could be removed.

Reply:
CC: @keith-zephyr, @jackrosenthal, @tari

Reply:
I prefer the first option.  This leverages the existing `io-channels` implementation in Zephyr.

Also in the first option, you don't need to provide the `adc-cmp` node parent.

Reply:
OK, I evaluated this when I was checking ITE VCMP case, it has only one register to set scan period, with second choice we could just use parent node to configure this.